sky/packages/sky/lib/src/gestures/events.dart

```diff
@@ -0,0 +1,114 @@
+import 'dart:ui' as ui;
```

Comment:
I think both are defensible designs. Immutable is the safest starting point, though.

If you make it immutable, make sure the constructor is "const" (and let all the fields be set via the default constructor with named arguments).